Allow setting the CPU profiling event for Java Async Profiler in the `pyroscope.java` component (#1258)
grafana/alloy: docs/sources/reference/components/pyroscope/pyroscope.java.md

`cpu`         | `bool`     | A flag to enable cpu profiling, using `itimer` async-profiler event.                                                                                                                                                                 | true    | no
`sample_rate` | `int`      | CPU profiling sample rate. It is converted from Hz to interval and passed as `-i` arg to async-profiler.                                                                                                                             | 100     | no
`alloc`       | `string`   | Allocation profiling sampling configuration  It is passed as `--alloc` arg to async-profiler.                                                                                                                                        | "512k"  | no
`lock`        | `string`   | Lock profiling sampling configuration. It is passed as `--lock` arg to async-profiler.                                                                                                                                               | "10ms"  | no

> `wall`        | `bool`     | Activates the `wall` mode when `cpu` profiling is enabled. This sample all threads equally every given period of time regardless of thread status: Running, Sleeping or Blocked. It is passed as `-e wall -t` arg to async-profiler. | false   | no

Reviewer: ```suggestion
`wall`        | `bool`     | Activates the `wall` mode when `cpu` profiling is enabled. This samples all threads equally every given period of time regardless of the thread status: Running, Sleeping, or Blocked. It is passed as an `-e wall -t` argument to the async-profiler. | false   | no
```

Should we move the extra information about `wall` to a spot below the table (like we do with other extended descriptions)?

`cpu`         | `bool`     | A flag to enable cpu profiling, using `itimer` async-profiler event by default.                             | true     | no
`sample_rate` | `int`      | CPU profiling sample rate. It is converted from Hz to interval and passed as an `-i` arg to async-profiler. | 100      | no
`alloc`       | `string`   | Allocation profiling sampling configuration  It is passed as an `--alloc` arg to async-profiler.            | "512k"   | no
`lock`        | `string`   | Lock profiling sampling configuration. It is passed as an `--lock` arg to async-profiler.                   | "10ms"   | no
`event`       | `string`   | Sets the CPU profiling event. Can be one of `itimer`, `cpu` or `wall`.                                      | "itimer" | no
`per_thread`  | `bool`     | Sets per thread mode on async profiler. It is passed as an `-t` arg to async-profiler.                      | false    | no

Refer to [profiler-options](https://github.com/async-profiler/async-profiler?tab=readme-ov-file#profiler-options) for more information about async-profiler configuration.

#### `event` argument

> Sets the CPU profiling event:
> * `itimer` - default, uses the [`setitimer(ITIMER_PROF)`](http://man7.org/linux/man-pages/man2/setitimer.2.html) 
>    syscall which generates a signal every time a process consumes CPU 
> * `cpu` - uses PMU-case sampling (like Intel PEBS or AMD IBS), can be more accurate than `itimer`, but it's not available in every platform.
> * `wall` - This sample all threads equally every given period of time regardless of thread status: Running, Sleeping or Blocked.
>    For instance, this can be helpful when profiling application start-up time or IO intensive processes.

Reviewer: ```suggestion
Sets the CPU profiling event:
* `itimer` - Default. Uses the [`setitimer(ITIMER_PROF)`](http://man7.org/linux/man-pages/man2/setitimer.2.html) 
   syscall, which generates a signal every time a process consumes CPU. 
* `cpu` - Uses PMU-case sampling (like Intel PEBS or AMD IBS), can be more accurate than `itimer`, but it's not available on every platform.
* `wall` - This samples all threads equally every given period of time regardless of thread status: Running, Sleeping, or Blocked.
   For example, this can be helpful when profiling application start-up time or IO-intensive processes.
```

#### `per_thread` argument  

> Sets per thread mode on async profiler. Profile threads separately, each stack trace will end with a frame that denotes a single thread.

Reviewer: `Profile threads separately`? or `Profiler`?

Author: Updated to `Threads are profiled` 🙂. Original description was from here:
https://github.com/async-profiler/async-profiler?tab=readme-ov-file#profiler-options

Reviewer: Ah that makes a LOT more sense now :-)

> Sets per thread mode on async profiler. Threads are profiled separately, each stack trace will end with a frame that denotes a single thread.

Reviewer: ```suggestion
Sets per thread mode on async profiler. Threads are profiled separately and each stack trace will end with a frame that denotes a single thread.
```
